### PR TITLE
Part II of II: Add the custom html pages theirselves to the addon

### PR DESCRIPTION
**Pull request by fliphess**
Hey Bas!

As mentioned in #12 a first setup to add custom templates. 

What is done:
- I've forked the (very old) [timoline](https://github.com/timoline/OTGW) repository and plan to add some edits later this week.  
- The template files are downloaded from github at build time and saved in /usr/share/otmonitor. 
- If the user chooses to enable the template feature, these files are just once copied to an accessible shared location (`/share/otmonitor/html`), from where they can be edited using a terminal or other addons like vscode or the file editor.
- At startup the editable files are copied to a subdir in the working directory where the binary is stored (in `/opt/html`) so they can be used instead of the default templates (https://github.com/hvxl/otmonitor/blob/master/otmonitor.vfs/otmonitor.tcl#L29)

Let me know what you think of this approach and if you like me to continue :)

### CODE REVIEW

**Comment by fliphess** [comment]
Glad that you mention it, I was thinking the same over a night of sleep :D 
I'll move the templates to this repo so we could prettify them in this repo 👍

**Comment by basnijholt** [comment]
Nice!

Does this always put files in `/share/otmonitor/html` even though a user might use the default template?

**Comment by fliphess** [comment]
It does that atm as I wanted to copy the files in a location where you can edit them yourself (to start simple). 

But for users that don't want to edit the files theirselves, that adds some clogging to their data directory which is not ideal.
I can easily add a "make the templates editable" option to the config.json and some extra conditionals to ensure files are only copied to `/share` if enabled in the configuration :) 

I'll add it this evening, I want to do some more testing as well :)

**Comment by basnijholt** [comment]
> I can easily add a "make the templates editable" option to the config.json and some extra conditionals to ensure files are only copied to /share if requested in the configuration :)

That sounds like a great idea!

**Comment by basnijholt** [comment]
@fliphess, let me know when it is done such that I can give it a final look over and merge it :)

File: otmonitor/rootfs/usr/share/otmonitor/html/navbar.inc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			<ul class="nav navbar-nav navbar-right">
+				<li><a href="status.html">Status</a></li>
+				<li><a href="graph.html">Graph</a></li>
+				<li><a href="messages.html">Messages</a></li>
+				<li><a href="summary.html">Statistics</a></li>
+                <li><a href="thermostat.html" target="_blank">Thermostat</a></li>		
+				<li class="dropdown">
+					<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Administration <span class="caret"></span></a>
+					<ul class="dropdown-menu">
+					<!--<li class="active"><a href="alert.html">Alert</a></li>
```
Comment (basnijholt) [inline otmonitor/rootfs/usr/share/otmonitor/html/navbar.inc:25]: Are these comments still useful?

File: otmonitor/rootfs/usr/share/otmonitor/html/js/dashboard.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+    function updateDash(){
+        $.ajax({
+            url: "/json",
+            dataType: "json",
+            cache: false,
+            success: function(json) {
+
+			//json = json;
```
Comment (basnijholt) [inline otmonitor/rootfs/usr/share/otmonitor/html/js/dashboard.js:37]: Do we leave these comments in?

File: otmonitor/rootfs/usr/share/otmonitor/html/navbar.inc (hunk 1 above)
Comment (fliphess) [reply otmonitor/rootfs/usr/share/otmonitor/html/navbar.inc:25]: No not really, they are original code and very ugly :) I'll strip right after fixing the merge conflicts I've got myself in :D

File: otmonitor/rootfs/usr/share/otmonitor/html/alert.html.tmpl [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+<div class="container">
+	<div class="row-content">
+<h1>Alert</h1>
+<div> 
+
+<form class="form-horizontal" role="form" enctype="multipart/form-data" method="post" action="alert.html">
+<div class="form-group">
+    <label class="control-label col-sm-2" for="alert,pressurehigh">Alert pressurehigh:</label>
+    <div class="col-sm-4">
+		<input type="text" class="form-control" name="alert,pressurehigh" value="$cfg(alert,pressurehigh)">
```
Comment (basnijholt) [inline otmonitor/rootfs/usr/share/otmonitor/html/alert.html.tmpl:25]: Would you mind to run these html templates through some "prettify" tool. Like https://www.freeformatter.com/html-formatter.html#ad-output would be good enough or just using VS Code. If too much work, I can do it ;)

Comment (fliphess) [reply otmonitor/rootfs/usr/share/otmonitor/html/alert.html.tmpl:25]: Absolutely :) I'm already doing some prettify as we type 👍

**Comment by fliphess** [comment]
Oh Wow! TCL/TK has it's own rules about identation :) 
I've to dive deeper into the templates as most of them now crash with a 500 error :) 

I've to spend some more time into this tomorrow :)

**Comment by fliphess** [comment]
And this one is finally ready too (I think..... I'm open to all suggestions)  :) 

That is ready as in "semi acceptable" ready I think: There is still a lot to customize (read: it's still ugly) but I'm pretty sure that some help from someone with a little more javascript skills and some taste for css can do wonders and this is a lot better than what it was before :) 💯

**Comment by basnijholt** [comment]
The HTML itself looks beautiful now 🥰

> Oh Wow! TCL/TK has it's own rules about identation :)

Did you manage to figure that one out?

**Comment by fliphess** [comment]
> The HTML itself looks beautiful now 🥰
Thanks! :)

> Did you manage to figure that one out?

Not completely :) I've found the [special characters](https://datacadamia.com/lang/tcl/special_character) for tcl/tk, but I couldn't really find any description yet on how to ident the special character lines. 

I continued putting the tcl statements (starting with `%`) at the beginning of the line, and the $vars work as with any programming language and I didn't have too much issues there after that :)

**Comment by fliphess** [comment]
Do you want to do some testing yourself on this one or shall I merge it?

**Comment by basnijholt** [comment]
I trust that it works. (Or if it's broken that you fix it 😄)

I'm going to try it out tomorrow! I'm excited!

**Pull request merged**